ska_conda/ska_builder.py

```diff
@@ -11,32 +11,54 @@ class SkaBuilder(object):
 
     def __init__(self, ska_root=None):
         if ska_root is None:
-            ska_root = "/data/acis/ska3_pkg/"
+            ska_root = "/tmp/ska3_pkg/"
         self.ska_root = ska_root
         self.ska_build_dir = os.path.join(self.ska_root, "builds")
         self.ska_src_dir = os.path.join(self.ska_root, "src")
         os.environ["SKA_TOP_SRC_DIR"] = self.ska_src_dir
 
-    def _clone_repo(self, name):
+    def _clone_repo(self, name, tag=None):
         if name == "ska":
             return
         print("Cloning source %s." % name)
         clone_path = os.path.join(self.ska_src_dir, name)
-        if os.path.exists(clone_path):
-            print("Source %s exists, skipping." % name)
-            return
-        yml = os.path.join(pkg_defs_path, name, "meta.yaml")
-        with open(yml) as f:
-            requires = False
-            while not requires:
-                line = f.readline().strip()
-                if line.startswith("path:"):
-                    requires = True
-            data = yaml.load(f)
-            url = data['about']['home']
-            git.Repo.clone_from(url, clone_path)
+        if not os.path.exists(clone_path):
+            # Try ssh first to avoid needing passwords for the private repos
+            # We could add these ssh strings to the meta.yaml for convenience
+            try:
+                git_ssh_path = 'git@github.com:sot/' + name + '.git'
+                repo = git.Repo.clone_from(git_ssh_path, clone_path)
+                assert not repo.bare
+            except:
+                yml = os.path.join(pkg_defs_path, name, "meta.yaml")
+                with open(yml) as f:
+                    requires = False
+                    while not requires:
+                        line = f.readline().strip()
+                        if line.startswith("path:"):
+                            requires = True
+                    data = yaml.load(f)
+                    url = data['about']['home']
+                repo = git.Repo.clone_from(url, clone_path)
+        else:
+            repo = git.Repo(clone_path)
+        assert not repo.is_dirty()
+        # I think we want the commit/tag with the most recent date, though
+        # if we actually want the most recently created tag, that would probably be
+        # tags = sorted(repo.tags, key=lambda t: t.tag.tagged_date)
+        # I suppose we could also use github to get the most recent release (not tag)
+        if tag is None:
+            tags = sorted(repo.tags, key=lambda t: t.commit.committed_datetime)
+            repo.git.checkout(tags[-1].name)
```

Comment:
This is obviously just one way of going about this.  I'm also wondering now if the {{ GIT_DESCRIBE_TAG }} method can break if the tag isn't formatted in a way that conda will like.  I note their docs say "Conda acknowledges PEP 440" but I don't really know what acknowledges means in that context.  We can obviously just update tags to work if they are broken, so that's just something to keep in mind.